File: BinarySearch/binSearch.cpp
#include <iostream>    
#include <algorithm>  
#include <vector>       
using namespace std;

int main () {
  int mylist[] = {1,3,6,10,15,4,23,12,11};
  vector<int> v(mylist,mylist+9);
  sort (v.begin(), v.end());
  int x;
  cout <<"What number are you looking for? ";
  cin >> x;
  if (binary_search (v.begin(), v.end(), x))
    cout << "The number you are looking for is in the list \n"; else cout << "The number you are looking for is not in the list.\n";  
  return 0;
}

File: BinarySearch/binSearch.py
def binarySearch(listgiven,item):
	f = 0
	l = len(listgiven)-1
	found = False
	while(not found and f<=l):
		m=(f+l)//2
		if listgiven[m]==item :
			found=True
		else:
			if item>listgiven[m]:
				f=m+1
			else:
				l=m-1
	return found
list = list(map(int, input("Enter list (separate each element by a space: ").split()))
list.sort()
numb = int(input("What number are you looking for?"))
if (binarySearch(list, numb)):
  print("It is in the list")
else:
  print("It is not in the list")
